forum/projects/serializers.py

```diff
@@ -0,0 +1,52 @@
+from rest_framework import serializers
+from .models import Project, Description
+
+
+class DescriptionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Description
+        fields = ['description']
+
+
+class ProjectSerializer(serializers.ModelSerializer):
+    """Nested serializer that handles both Project and its description"""
+
+    description = DescriptionSerializer()
+
+    class Meta:
+        model = Project
+        fields = '__all__'
+        read_only_fields = ['startup', 'is_completed']
+
+    def create(self, validated_data):
+        """Overridden method to create a project with description"""
+        description_data = validated_data.pop('description', None)
+        project = Project.objects.create(**validated_data)
+
+        # Handle description data if data exists
+        if description_data:
+            Description.objects.create(project=project, **description_data)
+        return project
+
+    def update(self, instance, validated_data):
+
+        """Overridden method that updates a project and it's description.
+         Or creates a new description if the old one doesn't exist"""
+
+        description_data = validated_data.pop('description', None)
+        for attr, value in validated_data.items():
+            setattr(instance, attr, value)
+        instance.save()
+
+        # Handle description data if data exists
+        if description_data:
+            if hasattr(instance, 'description'):
+                for attr, value in description_data.items():
+                    setattr(instance.description, attr, value)
+                instance.description.save()
+            else:
+                Description.objects.create(project=instance, **description_data)
```

Comment:
Ensure that description data meets specific criteria (e.g., length constraints, non-emptiness) if required.

Reply:
Added MaxLength validator in Description model, 2000 characters. Validation is now automatically handled by Description serializer.
MaxLengthValidator(5) was used just for test.
<img width="1380" alt="Screenshot 2024-12-21 at 22 53 46" src="https://github.com/user-attachments/assets/5c3d6b7e-254c-4aa8-b3b5-04f7ff1b1bf7" />